src/test/mockFactory.ts

```diff
@@ -334,15 +334,15 @@ export class MockFactory {
     };
   }
 
-  public static createTestFunctionMetadata(name: string): ServerlessAzureFunctionConfig {
+  public static createTestFunctionMetadata(name: string, xAzureSettings: boolean = true): ServerlessAzureFunctionConfig {
```

Comment:
Seems a little messy to add a new param to all these mock methods special for the xAzureSettings case.  Maybe consider creating new mock method for testing?